Lab/src/main/java/agh/ics/oop/Animal.java

```diff
@@ -1,20 +1,26 @@
 package agh.ics.oop;
+import java.util.ArrayList;
+import java.util.List;
 
 public class Animal {
     private Vector2d position;
+    private List<IPositionChangeObserver> observers = new ArrayList<>();
     private MapDirection direction;
-    private final  IWorldMap map;
-    //this() konstruktor w konstruktorze z mniejszego wiekszy
+    private final IWorldMap map;
     public Animal(IWorldMap map){
-        this.map = map;
-        position = new Vector2d(2,2);
-        direction = MapDirection.NORTH;
+    this(map,new Vector2d(2,2));
     }
     public Vector2d getPosition(){
         return this.position;
     }
     public Animal(IWorldMap map, Vector2d initialPosition){
         this.map= map;
+        if(this.map instanceof GrassField){
```

Comment:
to zupełnie niepotrzebne, wszystkie mapy powinny mieć opcję obserwowawnia - można rozszerzyć nawet interfejs `IWoldMap` i wtedy nie będzie tu problemu z rzutowaniem/sprawdzaniem typu.